content/ember/v3/deprecate-get-with-default.md

```diff
@@ -0,0 +1,78 @@
+---
+id: ember-metal.get-with-default
+title: Use Ember getter and explicitly check for undefined
+until: '4.0.0'
+since: '3.19'
+---
+
+Deprecate support for `getWithDefault` in Ember's Object module (@ember/object) – both the [function](https://api.emberjs.com/ember/release/functions/@ember%2Fobject/getWithDefault) and the [class method](https://api.emberjs.com/ember/release/classes/EmberObject/methods/getWithDefault?anchor=getWithDefault) – because its expected behaviour is confusing to Ember developers.
+
+- The API will only return the default value when the value of the property retrieved is `undefined`. This behaviour is often overlooked when using the function where a developer might expect that `null` or other _falsy_ values will also return the default value.
+- The native JavaScript [Nullish Coalescing Operator `??`](https://github.com/tc39/proposal-nullish-coalescing) could be used to handle this case if we also take `null` as a _falsey_ value to show the default value
+
+Before:
+
+```js
+import { getWithDefault } from '@ember/object';
+
+let result = getWithDefault(obj, 'some.key', defaultValue);
+```
+
+After:
+
+```js
+import { get } from '@ember/object';
+
+let result = get(obj, 'some.key');
+if (result === undefined) {
+  result = defaultValue;
+}
+```
+
+#### Using Nullish Coalescing Operator
+
+We cannot codemod directly into the nullish coalescing operator since the expected behaviour of `getWithDefault` is to only return the default value if it is strictly `undefined`. The nullish coalescing operator accepts either `null` or `undefined` to show the default value.
```

Comment:
Maybe add a link to nullish coalescing on MDN or something?

Reply:
Added!